docs/design/storagenode-graceful-exit.md

```diff
@@ -91,10 +92,169 @@ This process including the Storage nodes transferring their pieces to other node
 	- Receiving storage nodes should be uploaded to as if they were receiving any other upload (orders, etc).
 - We need to have the storage node keep track how its progress during graceful exit by calculating how much data it had when it started the graceful exit and keeping track of how much data it has deleted (data is deleted from the exiting node as it sends pieces to new nodes). This is an estimation since the storage node is likely holding more data than it will gracefully exit (garbage data). 
 
+## Implementation (MVP)
+
+#### Satellite
+- Update DBX - Add fields to `nodes` table
+  ```
+    model nodes (
+		...
+		field exit_loop_count           int
+		field exit_initiated_at         timestamp ( updateable )
+		field exit_completed_at         timestamp ( updateable )
+
+	}
+  ```
+- Add GetExitingNodeIds method to overlaycache. Returns node IDs where `exit_initiated_at` is not null and `exit_completed_at` is null.
+- Add GetExitedNodeIds method to overlaycache. Returns node IDs where `exit_initiated_at` is not null and `exit_completed_at` is not null. 
+- Update DBX - Add table exit_pieceinfos
+   ```
+	model exit_pieceinfo (
+		key node_id path
+
+		field node_id             blob
+		field path                blob
+		field piece_num           blob
+		field durability_ratio    float64
+		field queued_at           timestamp ( autoinsert )
+		field requested_at        timestamp ( updateable )
+		field failed_at           timestamp ( updateable )
+		field failed_status_code  int
+		field completed_at        timestamp ( updateable )
```

Comment:
Why do we keep completed pieces in the table? It's going to waste a lot of memory.